goldens/foo/lib/metadata.analyzer.json

```diff
@@ -0,0 +1,164 @@
+{
+  "uris": {
+    "package:foo/metadata.dart": {
+      "scopes": {
+        "Foo": {
+          "members": {
+            "": {
+              "requiredPositionalParameters": [],
+              "optionalPositionalParameters": [],
+              "namedParameters": [],
+              "properties": {
+                "isAbstract": false,
+                "isConstructor": true,
+                "isGetter": false,
+                "isField": false,
+                "isMethod": false,
+                "isStatic": false
+              }
+            }
+          },
+          "metadataAnnotations": [
+            {
+              "expression": {
+                "type": {
+                  "reference": {
+                    "type": "ClassReference",
+                    "value": {}
```

Comment:
How much information do you need for the references to have meaning? A name, uri, enclosing declaration name, ... ?

Reply:
The first obvious thing is that a macro needs to be able to identify its own annotation(s), which would "mostly work" with just name but needs name+uri to be correct.

I guess the two hardest use cases we have are the dependency injection framework example of aggregating "providers", and the general case of wanting to be able to output expressions into macro-generated code.

Reply:
I'll add these the `Reference`s.